src/hooks/useLifecycleLogger.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffectOnceWhen } from './useEffectOnceWhen';
+import { useUpdateEffect } from './useUpdateEffect';
+import { useWillUnmount } from './useWillUnmount';
+
+/**
+ * useLifecycleLogger hook
+ *
+ * @param componentName Name of the component
+ * @param rest
+ */
+const useLifecycleLogger = (componentName: string = 'Component', ...rest) => {
+  useEffectOnceWhen(() => {
```

Comment:
Just the one small change here. We need to use `useDidMount` here instead of `useEffectOnceWhen`